cmd/apprepository-controller/server/controller.go

```diff
@@ -154,21 +154,43 @@ func NewController(
 		log.Warningf("Error adding AppRepository event handler: %v", err)
 	}
 
+	controller.setBatchLister(conf.V1Beta1CronJobs, kubeInformerFactory)
+
+	return controller
+}
+
+// setBatchListener sets the specific batch listener based on the config, ie. either
+// a v1 batch listener or a v1beta1 batch listener.
+//
+// This allows users on 1.20 to continue to use the latest release.
+func (c *Controller) setBatchLister(useV1Beta1 bool, kubeInformerFactory kubeinformers.SharedInformerFactory) {
```

Comment:
Kind of makes me wonder if this flag is necessary at all. Could the code just iterate over both `v1` and `v1beta1` and query the k8s API to see which resource is defined in the cluster, and use that?

Reply:
Then you probably wouldn't need as many `if` checks/logic branches in the rest of the code.

Reply:
We could just iterate implicitly but that has drawbacks as well: we'd need to do that iteration at each call-site (where the `if` logic branches are, which could be extracted, yep), or just iterate once here and keep the conditionals, but more importantly, it'd become an *implicit* decision: I'm happy for this to be explicit with a branch in each part because we want to remove it again soon. It's not a supported feature (why I left the chart requirement unchanged).